File: CSharp/Yow.CoD.Finance.Domain/Services/CreateLoanCommandHandler.cs
using System.Threading.Tasks;
using Yow.CoD.Finance.Domain.Contracts;
using Yow.CoD.Finance.Domain.Model;

namespace Yow.CoD.Finance.Domain.Services
{
    public class CreateLoanCommandHandler : IHandler<CreateLoanCommand, Receipt>
    {
        private readonly IRepository<Loan> _repository;

        public CreateLoanCommandHandler(IRepository<Loan> repository)
        {
            _repository = repository;
        }

        public async Task<Receipt> Handle(CreateLoanCommand command)
        {
            var loan = await _repository.Get(command.AggregateId);
            loan.Create(command);
            await _repository.Save(loan);
            return new Receipt(loan.Id, loan.Version);
        }
    }
}

File: java/src/main/java/com/leecampbell/cod/domain/services/CreateLoanCommandHandler.java
package com.leecampbell.cod.domain.services;

import com.leecampbell.cod.domain.contracts.*;
import com.leecampbell.cod.domain.model.Loan;

public class CreateLoanCommandHandler implements CommandHandler<CreateLoanCommand, Receipt>
    {
        private final Repository _repository;

        public CreateLoanCommandHandler(Repository repository)
        {
            _repository = repository;
        }

        public Receipt handle(CreateLoanCommand command)
        {
            Loan loan = _repository.Get(command.aggregateId());
            loan.Create(command);
            _repository.Save(loan);
            return new Receipt(loan.id(), loan.version());
        }
    }